packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/services/auth_matrix_service.dart

```diff
@@ -0,0 +1,173 @@
+    required ActionRequest payload,
+    required String userData,
+  }) async {
+    _flowEvents
+        .add(await _authMatrixRepository.initiateAuthMatrix(payload: payload));
+    _endToEndId = payload.endToEndId;
+    _userData = userData;
+    await for (AuthMatrixResponse response in _flowEvents) {
+      switch (response.authZ) {
+        case AuthMatrixActionType.none:
+          _routerService.pop();
+          return;
+        case AuthMatrixActionType.pinAndOtp:
+        case AuthMatrixActionType.pinOnly:
+          await _navigateFunction(response, payload.endToEndId);
+      }
+    }
+    return;
+  }
+
+  ///Function used to navigate the user to the correct screen
+  /// based on the server response
+  Future<void> _navigateFunction(
+      AuthMatrixResponse response, String endToEndId) async {
+    switch (response.authZ) {
+      case AuthMatrixActionType.pinOnly:
+        await _routerService.go(
+            AuthMatrixPinBiometricsRoute(
+              endToEndId,
+            ),
+            response);
+      case AuthMatrixActionType.pinAndOtp:
+        await _routerService.go(
+            AuthMatrixOtpRoute(
+              endToEndId,
+            ),
+            response);
+      case AuthMatrixActionType.none:
+        break;
+    }
+  }
+
+  ///Function used to verify the auth matrix flow
+  ///Adds the response to the [_flowEvents] stream
+  Future<void> verifyAuthMethod(
+    AuthMatrixResponse authMatrixResponse,
+    String endToEndId,
+    String userData,
+    String code,
+  ) async {
+    _flowEvents.add(
+      await _authMatrixRepository
+          .verifyAuthMatrix(
+        payload: AuthMatrixVerify(
+          transactionId: authMatrixResponse.transactionId,
+          userData: userData,
+          payload: {'code': code},
+          action: authMatrixResponse.authZ,
+          endToEndId: endToEndId,
+        ),
+      )
+          .catchError(
+        (error, stacktrace) {
+          _flowEvents.addError(
+            AccessDeniedErrorModel(),
+          );
+          throw AccessDeniedErrorModel();
+        },
+      ),
+    );
+  }
+
+  ///Function used to cancel the auth matrix flow
+  ///Adds the error to the [_flowEvents] stream
+  Future<void> cancelAuthMatrix(
+    String endToEndId,
+    String transactionId,
+  ) async {
+    await _authMatrixRepository.cancelAuthMatrix(
+      AuthMatrixCancelModel(endToEndId, transactionId),
+    );
+    _flowEvents.addError(AuthMatrixCanceledErrorModel());
+  }
+
+  void dispose() {
+    _flowEvents.close();
+  }
+
+  @override
+  ///Function used to verify auth method once the user enters the OTP code
+  Future<dynamic> confirmPhoneCode(String code) async {
+    await verifyAuthMethod(_flowEvents.value, _endToEndId, _userData, code);
+    return true;
+  }
+
+  @override
+  Future<String> getFullPhoneNumber() async => '+38164 1234567';
+
+  @override
+  Future<String> updatePhoneNumber(String newNumber) async => '+38164 1234567';
```

Comment:
Even though they are dummy numbers, they still have a +381 prefix which is Serbian. When someone generates a project and sees that number, they might confuse it for a real number with some functionality. Better remove all references to any real life numbers, just to reduce confusion.

Reply:
Done